okapi-core/src/main/java/org/folio/okapi/managers/ProxyService.java

```diff
@@ -844,21 +850,24 @@ private void proxyRequestResponse(Iterator<ModuleInstance> it,
       clientRequest.onFailure(res -> proxyClientFailure(pc, mi, res));
       clientRequest.onSuccess(res -> {
         MetricsHelper.recordHttpClientResponse(sample, pc.getTenant(), res.statusCode(),
-            ctx.request().method().name(), mi);
-        fixupXOkapiToken(mi.getModuleDescriptor(), ctx.request().headers(), res.headers());
+            request.method().name(), mi);
+        fixupXOkapiToken(mi.getModuleDescriptor(), request.headers(), res.headers());
         Iterator<ModuleInstance> newIt = getNewIterator(it, mi, res.statusCode());
         if (res.getHeader(XOkapiHeaders.STOP) == null && newIt.hasNext()) {
           makeTraceHeader(mi, res.statusCode(), pc);
           relayToRequest(res, pc, mi);
-          final String ct = res.getHeader("Content-Type");
-          if (ct != null) {
-            ctx.request().headers().set("Content-Type", ct);
+          for (String header : FORWARD_HEADERS) {
+            String value = res.getHeader(header);
+            request.headers().remove(header);
+            if (value != null) {
+              request.headers().set(header, value);
+            }
```

Comment:
Seems you can combine those to one line `request.headers().set(header, res.getHeader(header));`

Reply:
Can I? Setting null is the same as not setting it at all?

Reply:
The doc says `If there is an existing header with the same name, it is removed. Setting a null value removes the entry.` So I think it is yes. Please verify though.

Reply:
Right you are.. Thx